plugins/inputs/odyssey/odyssey.go

```diff
@@ -0,0 +1,165 @@
+package odyssey
+
+import (
+	"bytes"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/postgresql"
+	_ "github.com/jackc/pgx/stdlib" // register driver
+)
+
+type Odyssey struct {
+	postgresql.Service
+	IncludeQuery []string `toml:"include_query"`
+}
+
+var ignoredColumns = map[string]bool{"user": true, "database": true, "pool_mode": true,
+	"avg_req": true, "avg_recv": true, "avg_sent": true, "avg_query": true,
+}
+
+var sampleConfig = `
+  ## specify address via a url matching:
+  ##   postgres://[console[:password]]@localhost[/dbname]\
+  ##       ?sslmode=[disable|verify-ca|verify-full]
+  ## or a simple string:
+  ##   host=localhost user=console password=... sslmode=... dbname=app_production
+  ##
+  ## All connection parameters are optional.
+  ##
+  address = "host=localhost user=postgresql sslmode=disable dbname=console port=6432"
+
+  ## Default queries - ['SHOW_POOLS', 'SHOW_ERRORS', 'SHOW_CLIENTS', 'SHOW_STATS']
+  ## A list of queries to include. If not specified, all the above listed queries are used.
+  include_query = []
+`
+
+func (p *Odyssey) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Odyssey) Description() string {
+	return "Read metrics from Odyssey"
+}
+
+func (p *Odyssey) Gather(acc telegraf.Accumulator) error {
+	var (
+		err     error
+		query   string
+		queries []string
+		columns []string
+	)
+
+	queries = []string{`SHOW POOLS`, `SHOW CLIENTS`, `SHOW ERRORS`, `SHOW STATS`}
```

Comment:
this should probably be a constant variable set at the top of the file

Reply:
I'd prefer to keep it here if only used in one location (which is the case here I think). The advantage is to have it close to the location where it is used. Otherwise this results in a lot of scrolling forth and back during review.

Reply:
Yeah that's fair.